$NetBSD: patch-aa,v 1.4 2008/04/24 07:45:49 joerg Exp $

Include pthread.h as thread stubs might be macros like on NetBSD.

Define conditional for cross-compiling support.

--- configure.ac.orig	2008-03-06 21:48:26.000000000 +0100
+++ configure.ac
@@ -180,7 +180,7 @@ esac
 AC_SUBST(XTHREADS)
 AC_SUBST(XUSE_MTSAFE_API)
 
-AC_CHECK_LIB(c, pthread_self, [thrstubs="no"], [thrstubs="yes"])
+AC_CHECK_DECLS([pthread_self], [thrstubs="no"], [thrstubs="yes"], [[#include <pthread.h>]])
 AM_CONDITIONAL(THRSTUBS, test x$thrstubs = xyes)
 
 dnl XXX incomplete, please fill this in
@@ -322,6 +322,8 @@ else
 fi
 AC_SUBST(XKBPROTO_REQUIRES)
 
+AM_CONDITIONAL(CROSS_COMPILING, [ test $cross_compiling = yes ])
+
 AC_FUNC_MMAP()
 composecache_default=$ac_cv_func_mmap_fixed_mapped
 AC_CHECK_FUNC(nl_langinfo, , [composecache_default=no])
